core/src/main/java/com/rudderstack/android/sdk/core/RudderClient.java

```diff
@@ -116,7 +116,7 @@ public static RudderClient getInstance(@NonNull Context context, @Nullable Strin
             RudderLogger.logVerbose("getInstance: instance null. creating instance");
             // assert writeKey is not null or empty
             if (TextUtils.isEmpty(writeKey)) {
-                throw new IllegalArgumentException("RudderClient: getInstance: writeKey can not be null or empty");
+                RudderLogger.logError("Invalid writeKey: Provided writeKey is empty");
```

Comment:
Ideally, we should abort the Initialization of the SDK when the write key is not provided by the user right? 

So can we update the log message to be more descriptive like, no write key provided hence SDK initialization is aborted ?

Reply:
Also, I feel we should return from here itself as the write key is not provided and SDK should be getting aborted immediately.

Reply:
Initialization is not aborted & will not aborted. We are not aborting initialisation. 
We can't return from here as the return expects a nonnull RudderClient's instance. And we can't change to nullable because this is public API.